tests/integration/synapseutils/test_synapseutils_sync.py

```diff
@@ -391,6 +368,311 @@ async def test_sync_to_synapse_activities_pointing_to_files(
                 assert len(file.activity.executed) == 1
                 assert file.activity.executed[0].target_id in file_ids
 
+    async def test_sync_to_synapse_activities_added_then_removed(
+        self, syn: Synapse, schedule_for_cleanup, project_model: Project
+    ) -> None:
+        # GIVEN a folder to sync to
+        folder = await Folder(
+            name=str(uuid.uuid4()), parent_id=project_model.id
+        ).store_async()
+        schedule_for_cleanup(folder.id)
+
+        # AND temporary file on disk:
+        temp_file = utils.make_bogus_uuid_file()
+        schedule_for_cleanup(temp_file)
+
+        # AND A manifest file with the paths to the temp files exists
+        df = pd.DataFrame(
+            {
+                "path": [temp_file],
+                "parent": folder.id,
+                "used": SYNAPSE_URL,
+                "executed": "",
+                "activityName": BOGUS_ACTIVITY,
+                "activityDescription": BOGUS_DESCRIPTION,
+            }
+        )
+        # Write the df to the file:
+        file_name = write_df_to_tsv(df, schedule_for_cleanup)
+
+        # WHEN I sync the manifest to Synapse
+        synapseutils.syncToSynapse(syn, file_name, sendMessages=SEND_MESSAGE, retries=2)
+
+        # THEN I expect that the folder has all of the files
+        await folder.sync_from_synapse_async(download_file=False)
+        assert len(folder.files) == 1
+
+        # AND the file is the one we uploaded
+        assert folder.files[0].version_number == 1
+        assert folder.files[0].path == temp_file
+        assert folder.files[0].activity.name == BOGUS_ACTIVITY
+        assert folder.files[0].activity.description == BOGUS_DESCRIPTION
+        assert len(folder.files[0].activity.used) == 1
+        assert folder.files[0].activity.used[0].url == SYNAPSE_URL
+
+        # WHEN I update the manifest file to remove the activities
```

Comment:
Just wanted to confirm that this is existing behavior for the provenance to be removed when there are blank values in used / executed

Reply:
It's slightly different. If used/executed it not specified it just won't apply a new Activity to the new version of your file. 

In this case I'm removing the Activity from your current Entity version if you don't specify it. If you have a new version 2, the original Activity on version 1 still exists, and no Activity exists on 2.

Since this is slightly different, See the other thread about my boolean idea to extend the things you can do with this function.